app/src/org/commcare/tasks/FormRecordLoaderTask.java

```diff
@@ -260,4 +274,83 @@ public boolean registerPriority(FormRecord record) {
         }
     }
 
+    private static String getTitleFromSession(AndroidSessionWrapper androidSessionWrapper) {
+        //TODO: Most of this mimicks what we need to do in entrydetail activity, remove it from there
+        //and generalize the walking
+
+        // get a copy of the session
+        CommCareSession session = new CommCareSession(androidSessionWrapper.getSession());

```

Comment:
See https://github.com/dimagi/commcare-odk/pull/1187 , which addresses https://github.com/dimagi/commcare-odk/pull/1187/files#diff-eebcc4a423830db1c3a3aa4d9d827c6cL201 which seems pretty sketch to me, even if it is behaviorally okay.